cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -766,16 +799,45 @@ private String pgId(CQLTermNode node) throws QueryValidationException {
    * @throws QueryValidationException
    */
   private String index2sql(String index, CQLTermNode node) throws QueryValidationException {
+    IndexTextAndJsonValues vals = getIndexTextAndJsonValues(index);
+    DbIndex dbIndex = DbSchemaUtils.getDbIndex(dbTable, index);
+
+    CqlModifiers cqlModifiers = new CqlModifiers(node);
+    List<Modifier> relationModifiers = cqlModifiers.getRelationModifiers();
+    if (relationModifiers.isEmpty()) {
+      return indexNode(index, node, vals, dbIndex, cqlModifiers);
+    } else if (relationModifiers.size() > 1) {
+      throw new QueryValidationException("CQL: Only one @-relation modifier supported");
+    } else {
+      Modifier relationModifier = relationModifiers.get(0);
```

Comment:
Using guard clauses reduces the indentation of the else-clause and the complexity, and increases readability:

    if (elationModifiers.isEmpty()) {
      return indexNode(index, node, vals, dbIndex, cqlModifiers);
    }
    if (relationModifiers.size() > 1) {
      throw new QueryValidationException("CQL: Only one @-relation modifier supported");
    }
    Modifier relationModifier = relationModifiers.get(0);

Reply:
OK. But I'll try to generalize it instead.. Was not really happy with only one relation modifier.